application-diagram-ui/src/main/resources/Diagram/DiagramMacro.xml

```diff
@@ -263,20 +263,27 @@
     #if ($diagramObj)
       #if ($xcontext.action == 'edit' || $xcontext.action == 'export' || $request.xpage == 'print'
           || $xcontext.macro.params.cached == 'true')
-        #set ($fileName = 'diagram.svg')
-        #set ($pngFileName = 'diagram.png')
-        #if ($xcontext.action == 'export' &amp;&amp; $diagramObj.getValue('exportUsingSVG') == 0
-            &amp;&amp; $diagram.getAttachment($pngFileName))
-          #set ($fileName = $pngFileName)
-          #set ($diagramURL = $diagram.getAttachmentURL($fileName, 'download', "v=$!diagram.version"))
-          #set ($output = "{{html clean='false'}}&lt;img src='$diagramURL' alt='$escapetool.xml($diagramTitle)'/&gt;{{/html}}")
-        #elseif ($xcontext.action == 'export')
-          #set ($diagram = $xwiki.getDocument($reference))
-          #set ($output = "{{html}}&lt;div&gt;$diagram.getAttachment('diagram.svg').getContentAsString()&lt;/div&gt;{{/html}}")
+        #set ($pageIndex =  $xcontext.macro.params.pageIndex)
+        #if ($pageIndex &gt;= 1)
+          #getAttachmentName($pageIndex)
+          #set ($fileName = "${attachmentName}.svg")
```

Comment:
If the selected index does not exist, I'd add a warning similar to the negative page index one.